servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -325,8 +325,17 @@ public EntriesResponse getEntries(String namedRef, EntriesParams params)
                           .name(fromKey(key.getValue()))
                           .type((Type) key.getType())
                           .build())) {
-        entries =
-            filterEntries(s, params.queryExpression()).collect(ImmutableList.toImmutableList());
+        Stream<EntriesResponse.Entry> entriesStream =
+            filterEntries(entryStream, params.queryExpression());
+        if (params.namespaceDepth() != null && params.namespaceDepth() > 0) {
+          entries =
```

Comment:
nit: this can be:
```
entriesStream = entriesStream
                  .map(e -> truncate(e, params.namespaceDepth()))
                  .distinct();
```
and remove the `else`

Reply:
done